portfolio/src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -14,38 +14,90 @@
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.Log;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
 
 @WebServlet("/user")
 public class UserServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html");
-    boolean loggedIn = false;
-
+    Query query = new Query("Log");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
     UserService userService = UserServiceFactory.getUserService();
-    if (userService.isUserLoggedIn()) {
-      String userEmail = userService.getCurrentUser().getEmail();
-      String urlToRedirectToAfterUserLogsOut = "/";
-      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
-      
-      loggedIn = true;
-      response.getWriter().println("Logged in, " + userEmail + "!");
-      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
-    } else {
-      String urlToRedirectToAfterUserLogsIn = "/";
-      String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+
+    
+    
+    List<Log> Logs = new ArrayList<>();
+
+
+    for (Entity entity : results.asIterable()) {//iterate through the logs
+        //get all the properties
+        if (userService.isUserLoggedIn()) {
+          String status = "In";
+          String userEmail = userService.getCurrentUser().getEmail();
+          String urlToRedirectToAfterUserLogsOut = "/";
+          String loginUrl = "";
+          String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
       
-      loggedIn = false;
-      response.getWriter().println("Not logged in");
-      response.getWriter().println("<p>Login <a href=\"" + loginUrl + "\">here</a>.</p>");
+        } 
+        else {
+          String status = "Out";
+          String urlToRedirectToAfterUserLogsIn = "/";
+          String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+          String logoutUrl = "";
+        }
+         
+         //create new log object
+          Log log = new Log(status, loginUrl, logoutUrl, userEmail);
+         
+         //add to Logs
+         Logs.add(log);
     }
+
+
+    Gson gson = new Gson();
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(Logs));
```

Comment:
Is there a way to send a response that is Logs as Json without printing it out? Like I could select parts to print out in the Javascript file (ie: urls or emails)?

Reply:
The `response.getWriter().println(` piece is the part that builds the response.  If you imagine the response as an electronic text file, the `println` here prints the json to that electronic text file (which is then sent back to the client)